bin2tif/readme.md

```diff
@@ -14,6 +15,80 @@ _Output_
 
   - The dataset containing the left/right binary images will get left/right JPG and TIFF images
 
+## Algorithm
+
+### Algorithm Description
+
+1. Convert raw data to 3 channels color image
+
+   Stereo RGB camera use a single charge-coupled device(CCD) sensor, with the CCD pixels preceded in the optical path by a color filter array(CFA) in a Bayer mosaic pattern. For each 2x2 set of pixels, two diagonally opposed pixels have green filters, and the other two have red and blue filters. We are assuming it’s in a GBRG ordering, then use bilinear interpolation to do the demosaicing. That means three color planes are independently interpolated using symmetric bilinear interpolation from the nearest neighbors of the same color.
+
+**Reference: Malvar, H.S., L. He, and R. Cutler, High quality linear interpolation for demosaicing of Bayer-patterned color images. ICASPP, Volume 34, Issue 11, pp. 2274-2282, May 2004.**
+
+2. Steps for geo-referencing bounding box to each image.
+
+   a. Get image shape from metadata,
+
+   b. Get camera center position from metadata
+
+   c. Compute field of view for the image
+   ```
+       i. The JSON data reports the camera field of view as "the field of view for a scene 2 meters away is: “0.749m x 1.015m"
+
+       ii. Predict fov for each image should be:
+
+                  fix_fov = fov_in_2_meter*(camera_height/2)
+
+       iii. In implementing the stitching process, we required two magic numbers that are computed experimentally to get the correct geometric alignment.  Our experimentally determined values are:
+
+           HEIGHT_MAGIC_NUMBER = 1.64
+
+           PREDICT_MAGIC_SLOPE = 0.574
+
+           And they are used in the following way:
+
+		            predict_plant_height = PREDICT_MAGIC_SLOPE * camHeight
+
+		            camH_fix = camHeight + HEIGHT_MAGIC_NUMBER - predict_plant_height
+
+		            fix_fov_x = fov_x*(camH_fix/2)
+
+		            fix_fov_y = fov_y*(camH_fix/2)
+   ```
+   d. Compute geo-reference bounding box
+   ```
+        i. Convert coordinates from Scanalyzer to MAC coordinates using formula from [https://terraref.gitbooks.io/terraref-documentation/content/user/geospatial-information.html](https://terraref.gitbooks.io/terraref-documentation/content/user/geospatial-information.html)
+
+        ii. Use utm tools to convert coordinates from MAC to lat/lon
+   ```
+   e. Using osgeo.gdal, associate with geo-bounding box, create geotiff.
+
+### Parameters
+
+HEIGHT_MAGIC_NUMBER and PREDICT_MAGIC_SLOPE were applied when we estimate the field of view, this is a testing based number.
```

Comment:
@ZongyangLi when you say in the README "HEIGHT_MAGIC_NUMBER and PREDICT_MAGIC_SLOPE were applied when we estimate the field of view, this is a testing based number." ... is it correct to say that these numbers are empirically derived based on some calibration? Do you have additional information about how these numbers are derived?

Reply:
Background: when I use FOV in metadata, I am not able to get a well stitched full field map.

Steps for creating these MAGIC NUMBER:
1. Make an assumption that the real FOV can be derived as a linear function base on the given FOV.
2. Randomly pick several different days stereoTop RGB raw data in different camera height, try to get a best stitched full field map for each day empirically(by personal judgement), and record all the FOV for each day.
3. Apply a linear regression to output HEIGHT_MAGIC_NUMBER and PREDICT_MAGIC_SLOPE